java/ql/lib/semmle/code/java/dataflow/ExternalFlow.qll

```diff
@@ -430,6 +430,7 @@ private Element interpretElement0(
 }
 
 /** Gets the source/sink/summary/neutral element corresponding to the supplied parameters. */
+cached
 Element interpretElement(
```

Comment:
I suspect that this is actually not the right thing to cache. I checked where it is used, and I think it ultimately boils down to the `TDataFlowCallable` newtype not being cached.

Reply:
We probably also need to cache `viableCallableExt` in `DataFlowImplCommon.qll`.

Reply:
The latter is done here: https://github.com/github/codeql/pull/15582